plugin/middleware.go

```diff
@@ -53,12 +53,14 @@ type SpecGetter func(string) *MiddlewareSpec
 
 // Registry contains currently registered middlewares and used to support pluggable middlewares across all modules of the vulcand
 type Registry struct {
-	specs []*MiddlewareSpec
+	specs              []*MiddlewareSpec
+	notFoundMiddleware Middleware
```

Comment:
Can you rename it to notFound for simplicity as the Middleware is already enforced by the type